### PR TITLE
[Datastore] Rename `DatastoreProfileAzureBlob` 'bucket' parameter to 'container'

### PR DESCRIPTION
**Pull request by alxtkr77**
https://iguazio.atlassian.net/browse/ML-6326

### CODE REVIEW

**Review by gtopper — APPROVED** [review]

File: mlrun/datastore/datastore_profile.py [hunk 1]
```diff
@@ -344,24 +344,24 @@ class DatastoreProfileAzureBlob(DatastoreProfile):
     client_secret: typing.Optional[str] = None
     sas_token: typing.Optional[str] = None
     credential: typing.Optional[str] = None
-    bucket: typing.Optional[str] = None
+    container: typing.Optional[str] = None
 
-    @pydantic.validator("bucket")
-    def check_bucket(cls, v):
+    @pydantic.validator("container")
+    def check_container(cls, v):
```
Comment (gtopper) [inline mlrun/datastore/datastore_profile.py]: Need to add `@classmethod` or change `cls` -> `self`.
(all occurrences)

Comment (alxtkr77) [reply mlrun/datastore/datastore_profile.py]: Shouldn't we enforce it in lint though?

Comment (gtopper) [reply mlrun/datastore/datastore_profile.py]: @jond01

**Review by gtopper — APPROVED** [review]

File: mlrun/datastore/datastore_profile.py (hunk 1 above)
Comment (jond01) [reply mlrun/datastore/datastore_profile.py]: The relevant rules are:
* N804: https://docs.astral.sh/ruff/rules/invalid-first-argument-name-for-class-method/
* N805: https://docs.astral.sh/ruff/rules/invalid-first-argument-name-for-method/

They are a part of the pep8-naming rules (`N`), which are not selected in mlrun. I encourage you to add it 🙂

**Pull request merged**